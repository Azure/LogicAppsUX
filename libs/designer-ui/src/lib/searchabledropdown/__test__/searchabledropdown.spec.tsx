import type { SearchableDropdownProps } from '..';
import { SearchableDropdown } from '..';
import renderer from 'react-test-renderer';

// vi.mock('@fluentui/react', () => ({
//   ...vi.requireActual('@fluentui/react'),
//   Dropdown: 'Dropdown',
// }));
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';
describe('lib/searchabledropdown', () => {
  it('should create a dropdown without a search box for 3 options', () => {
    const props: SearchableDropdownProps = {
      dropdownProps: {
        multiSelect: true,
        placeholder: 'Select an option',
        options: [
          { key: 'foo', text: 'Foo' },
          { key: 'bar', text: 'Bar' },
          { key: 'baz', text: 'Baz' },
        ],
      },
      onItemSelectionChanged: vi.fn(),
    };
    const component = <SearchableDropdown {...props} />;
    const renderedComponent = renderer.create(component).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should create a dropdown with a search box for 5 options', () => {
    const props: SearchableDropdownProps = {
      dropdownProps: {
        multiSelect: true,
        placeholder: 'Select an option',
        options: [
          { key: 'foo', text: 'Foo' },
          { key: 'bar', text: 'Bar' },
          { key: 'baz', text: 'Baz' },
          { key: 'qux', text: 'Qux' },
          { key: 'qoz', text: 'Qoz' },
        ],
      },
      onItemSelectionChanged: vi.fn(),
    };
    const component = <SearchableDropdown {...props} />;
    const renderedComponent = renderer.create(component).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
