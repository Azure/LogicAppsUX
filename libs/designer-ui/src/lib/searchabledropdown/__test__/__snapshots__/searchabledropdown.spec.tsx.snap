// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/searchabledropdown should create a dropdown with a search box for 5 options 1`] = `
<Dropdown
  className="msla-searchable-dropdown"
  multiSelect={true}
  onChange={[Function]}
  onDismiss={[Function]}
  onRenderItem={[Function]}
  options={
    [
      {
        "itemType": 2,
        "key": "FilterHeader",
        "text": "",
      },
      {
        "itemType": 1,
        "key": "FilterDivider",
        "text": "-",
      },
      {
        "key": "foo",
        "text": "Foo",
      },
      {
        "key": "bar",
        "text": "Bar",
      },
      {
        "key": "baz",
        "text": "Baz",
      },
      {
        "key": "qux",
        "text": "Qux",
      },
      {
        "key": "qoz",
        "text": "Qoz",
      },
    ]
  }
  placeholder="Select an option"
  selectedKeys={[]}
/>
`;

exports[`lib/searchabledropdown should create a dropdown without a search box for 3 options 1`] = `
<Dropdown
  className="msla-searchable-dropdown"
  multiSelect={true}
  onChange={[Function]}
  onDismiss={[Function]}
  onRenderItem={[Function]}
  options={
    [
      {
        "key": "foo",
        "text": "Foo",
      },
      {
        "key": "bar",
        "text": "Bar",
      },
      {
        "key": "baz",
        "text": "Baz",
      },
    ]
  }
  placeholder="Select an option"
  selectedKeys={[]}
/>
`;
